internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -53,6 +64,79 @@ func (s *Server) SetUpSession(stream s2av2pb.S2AService_SetUpSessionServer) erro
 	}
 }
 
+// TODO(rmehta19): consider also returning an error with this function.
+func validatePeerCertificateChain(req *s2av2pb.ValidatePeerCertificateChainReq) (*s2av2pb.SessionResp, error) {
+	switch x := req.PeerOneof.(type) {
+	case *s2av2pb.ValidatePeerCertificateChainReq_ClientPeer_:
+		// TODO(rmehta19): Handle Verification Mode(UNSPECIFIED, SPIFFE, or CONNECT_TO_GOOGLE).
+		derCertChain := req.GetClientPeer().CertificateChain
+		if len(derCertChain) == 0 {
+			// TODO(rmehta19): Is it an error for peer to have empty cert chain(I think
+			// so, because we set ClientAuth as require request and verify)? If so,
+			// (server requires nonempty certificate chain) will error be raised
+			// for us by some existing API function way before this peer verification?
+			err := errors.New("Client Peer Verification failed: client cert chain is empty.")
+			return buildValidatePeerCertificateChainSessionResp(400, err.Error(), s2av2pb.ValidatePeerCertificateChainResp_FAILURE, err.Error()), err
```

Comment:
Why 400?

Reply:
Is this an HTTP status code? In general, we should use gRPC status codes rather than HTTP status codes.

https://developers.google.com/maps-booking/reference/grpc-api/status_codes

Reply:
Yes, I was using HTTP codes, I'll switch over to the gRPC ones, thanks for pointing this out!